cfgov/v1/tests/test_password_policy.py

```diff
@@ -40,47 +41,55 @@ def test_good_passwords(self):
-    def setUp(self):
-        user = User(username='testuser', password=make_password('password')) 
-        user.save()
+    def get_user(self, last_password='password', pw_locked_until=None):
+        encrypted_password = make_password(last_password)
 
-    def tearDown(self):
-        User.objects.get(username='testuser').delete()
+        if not pw_locked_until:
+            pw_locked_until = timezone.now()
 
-    def get_user(self):
-        return User.objects.get(username='testuser')
+        last_password_change = PasswordHistoryItem(
+            encrypted_password=encrypted_password,
+            locked_until=pw_locked_until
+        )
 
+        password_history = Mock(spec=PasswordHistoryItem.objects)
+        password_history.latest.return_value = last_password_change
+        password_history.order_by.return_value = [last_password_change]
 
-class TestMinimumPasswordAge(TestWithUser):
+        return Mock(
+            username='testuser',
+            password=encrypted_password,
+            passwordhistoryitem_set=password_history
+        )
 
-    def test_too_soon(self):
-        user = self.get_user()
-        history_item = user.passwordhistoryitem_set.latest()
-        current_locked_until = history_item.locked_until
-        new_locked_until = current_locked_until.replace(year=datetime.MAXYEAR)
-        history_item.locked_until = new_locked_until
-        history_item.save()
-        with self.assertRaises(ValidationError):
-            password_policy.validate_password_age(user)
-
-    def test_old_enough_to_change(self):
-        user = self.get_user()
-        history_item = user.passwordhistoryitem_set.get()
-
-        current_lock_expires = history_item.locked_until
-        new_expiration = current_lock_expires.replace(year=2013)
-        history_item.locked_until = new_expiration
-        history_item.save()
 
+class TestMinimumPasswordAge(TestWithUser):
+    def test_can_set_password_if_unlocked(self):
+        yesterday = timezone.now() - timedelta(days=1)
+        user = self.get_user(pw_locked_until=yesterday)
         password_policy.validate_password_age(user)
 
+    def test_cant_set_password_if_locked(self):
+        tomorrow = timezone.now() + timedelta(days=1)
+        user = self.get_user(pw_locked_until=tomorrow)
+        self.assertRaises(
+            ValidationError,
+            password_policy.validate_password_age,
+            user
+        )
+
 
 class TestPasswordReusePolicy(TestWithUser):
-    def test_reuse_password(self):
-        user = self.get_user()
-        with self.assertRaises(ValidationError):
-            password_policy.validate_password_history(user, 'password')
-
-    def test_new_password(self):
-        user = self.get_user()
-        password_policy.validate_password_history(user, 'new_password')
+    def test_can_set_new_password(self):
+        user = self.get_user('password1')
+        password_policy.validate_password_history(user, 'password2')
```

Comment:
can you pass in the parameters explicitly here, so `self.get_user(last_password='password1')` ?